frontend/src/component/AddVisitCountry.tsx

```diff
@@ -1,43 +1,100 @@
         }
+        const newPlusOldCountries = [...props.loggedInTraveller.visitedCountries, ...selectedCountries]
+        const copyLoggedInTraveller = props.loggedInTraveller
+        copyLoggedInTraveller.visitedCountries = newPlusOldCountries
+
+        updateUser(props.loggedInTraveller.id, copyLoggedInTraveller)
+            .then(() => {
+                props.loggedInTraveller.visitedCountries = newPlusOldCountries
+                navigate("/overview")
+            })
+            .catch(console.error)
+    }
+
+    function handleRemoveCountryFromList(countryId: string) {
+
+        setSelectedCountries((prevState) => {
+            return prevState.filter((country) => country.threeLetterCode !== countryId);
+        })
+    }
+
+    function addCountryToList(event: SyntheticEvent<Element, Event>, value: Country | null) {
         if (value) {
-            const copyVisitedCountries = [...props.loggedInTraveller.visitedCountries,value]
-            const copyLoggedInTraveller = props.loggedInTraveller
-            copyLoggedInTraveller.visitedCountries = copyVisitedCountries
-
-            updateUser(props.loggedInTraveller.id, copyLoggedInTraveller)
-                .then(() => {
-                    props.loggedInTraveller.visitedCountries= copyVisitedCountries
-                    navigate("/overview")
-                })
-                .catch(console.error)
+            setSelectedCountries((prevState) => {
+                return [...prevState, value]
+            })
         }
+    }
+
+    const buttonDisabled: boolean = selectedCountries.length < 1
+
+    if (countries) {
+        return (<Box display={"flex"}
+                     width={'100%'}
+                     justifyContent={"center"}
+                     marginTop={20}
+                     flexDirection={"column"}
+                     flexWrap={"wrap"}
+                     alignItems={"center"}
+            >
+                <Autocomplete sx={{width: "80%"}} onChange={addCountryToList}
+                              renderInput={(params) =>
+                                  <TextField {...params} label={"Choose a country"}/>}
+                              getOptionLabel={(option) => option.name + " " + option.flag}
+                              options={countries}
+                />
+                <Box marginTop={3}>
+                    <Typography textAlign={"center"}
+                                variant={"subtitle1"}>
+                        Selected countries:
+                    </Typography>
+
+                    {selectedCountries.map((selectedCountry) =>
+                        <Chip key={selectedCountry.threeLetterCode}
+                              label={selectedCountry.name + " " + selectedCountry.flag}
+                              color={"secondary"}
+                              onDelete={() => handleRemoveCountryFromList(selectedCountry.threeLetterCode)}></Chip>
+                    )}
+                </Box>
```

Comment:
Pro tip: You could wrap all the `Box` elements in subcomponents too. This would make understanding their purpose much easier!